### PR TITLE
Add Nanosaur II: Hatchling

### PR DESCRIPTION
**Pull request by jorio**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(AppStream metadata files are in upstream)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.jor.nanosaur2.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        commit: 589931b7475d2efa8ee64febd874773f0003f236
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+      - type: file
+        path: nanosaur2_launcher.sh
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    no-make-install: true
+    build-commands:
```
Comment (hfiguiere) [inline io.jor.nanosaur2.yml:30]:
```suggestion
    post-install:
```

File: io.jor.nanosaur2.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+  - --share=ipc
+
+modules:
+  - name: nanosaur2
+    buildsystem: cmake
+    sources:
```
Comment (hfiguiere) [inline io.jor.nanosaur2.yml:17]: sources are usually the last in a module.

File: io.jor.nanosaur2.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+  - --share=ipc
+
+modules:
+  - name: nanosaur2
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.jor.nanosaur2.yml:16]: cmake-ninja?

File: io.jor.nanosaur2.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+      - type: file
+        path: nanosaur2_launcher.sh
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    no-make-install: true
+    build-commands:
+      - install -D nanosaur2_launcher.sh ${FLATPAK_DEST}/bin/Nanosaur2
```
Comment (hfiguiere) [inline io.jor.nanosaur2.yml:31]:
```suggestion
      - install -Dm755 nanosaur2_launcher.sh ${FLATPAK_DEST}/bin/Nanosaur2
```

**Comment by hfiguiere** [comment]
bot, build  io.jor.nanosaur2

**Comment by hfiguiere** [comment]
bot, build io.jor.nanosaur2

File: io.jor.nanosaur2.yml (hunk 3 above)
Comment (jorio) [reply io.jor.nanosaur2.yml:16]: cmake-ninja works too, if that's preferred over plain cmake

File: io.jor.nanosaur2.yml (hunk 2 above)
Comment (jorio) [reply io.jor.nanosaur2.yml:17]: OK, will move it

File: io.jor.nanosaur2.yml (hunk 3 above)
Comment (hfiguiere) [reply io.jor.nanosaur2.yml:16]: it's faster.
`cmake` only exists for when the Ninja backend doesn't work.

Comment (jorio) [reply io.jor.nanosaur2.yml]: OK! Implemented in [684d9c0](https://github.com/flathub/flathub/pull/3885/commits/684d9c02a13e3a9cfaea9d9f8fce8096ba1a5190)

File: io.jor.nanosaur2.yml (hunk 4 above)
Comment (jorio) [reply io.jor.nanosaur2.yml]: Done in [684d9c0](https://github.com/flathub/flathub/pull/3885/commits/684d9c02a13e3a9cfaea9d9f8fce8096ba1a5190)

File: io.jor.nanosaur2.yml (hunk 2 above)
Comment (jorio) [reply io.jor.nanosaur2.yml:28]: Done in [684d9c0](https://github.com/flathub/flathub/pull/3885/commits/684d9c02a13e3a9cfaea9d9f8fce8096ba1a5190)

File: io.jor.nanosaur2.yml (hunk 1 above)
Comment (jorio) [reply io.jor.nanosaur2.yml]: Done in [684d9c0](https://github.com/flathub/flathub/pull/3885/commits/684d9c02a13e3a9cfaea9d9f8fce8096ba1a5190)

**Comment by barthalion** [comment]
bot, build io.jor.nanosaur2

**Pull request closed without merging**